src/test/java/com/endava/TestCreativeAndUXPage.java

```diff
@@ -15,53 +15,167 @@
-	}
+    @BeforeTest
+    @Parameters({"browser"})
+    public void setUp(String browser) {
+        WebDriverWrapper.setUpDriver(browser);
+    }
 
-	@BeforeMethod
-	@Parameters({ "browser" })
-	public void openBrowser(String browser) {
-		homePage = new HomePage(WebDriverWrapper.createDriver(browser));
-		homePage.open();
-	}
+    @BeforeMethod
+    @Parameters({"browser"})
+    public void openBrowser(String browser) {
+        homePage = new HomePage(WebDriverWrapper.createDriver(browser));
+        homePage.open();
+    }
+
+    /**
+     * Go to endava.com. Scroll down to the page footer and from the "Services"
+     * section click on "Creative & UX". Validate URL has changed. Scroll down to
+     * the "Contact us" area. Populate Country/Region and Email.
+	 * Validate Email Address and Country/Region name are correct. Click
+     * on "Contact US" button. Validate that for all mandatory fields there is a
+     * warning message if it's not populated (e.g. "Please enter First Name").
+     */
+    @Test
+    public void testCreativeAndUXPage1() {
+        logger.info("Test testCreativeAndUXPage start");
+
+        creativeAndUXPage = homePage.openCreativeAndUXPage();
+        creativeAndUXPage.assertUrlEndsWith("Creative-And-UX");
+        creativeAndUXPage.clickOnTypeOfMessage();
+        creativeAndUXPage.enterEmail("indira.gandhi@delhi.com");
+        creativeAndUXPage.enterCountry("India");
+        creativeAndUXPage.validateEmail("indira.gandhi@delhi.com");
+        creativeAndUXPage.validateCountry("India");
+        creativeAndUXPage.clickOnCotactUsButton();
+        creativeAndUXPage.assertFirstNameWarningMessageIsShown();
+
+        logger.info("Test testCreativeAndUXPage end");
+    }
+
+
+	/**
+	 * Go to endava.com. Scroll down to the page footer and from the "Services"
+	 * section click on "Creative & UX". Validate URL has changed. Scroll down to
+	 * the "Contact us" area. Populate First name, Country/Region and Email.
+	 * Validate first name, Email Address and Country/Region name are correct. Click
+	 * on "Contact US" button. Validate that for all mandatory fields there is a
+	 * warning message if it's not populated (e.g. "Please enter Last Name").
+	 */
+    @Test
+    public void testCreativeAndUXPage2() {
+        logger.info("Test testCreativeAndUXPage start");
+
+        creativeAndUXPage = homePage.openCreativeAndUXPage();
+        creativeAndUXPage.assertUrlEndsWith("Creative-And-UX");
+        creativeAndUXPage.clickOnTypeOfMessage();
+        creativeAndUXPage.enterEmail("indira.gandhi@delhi.com");
+        creativeAndUXPage.enterCountry("India");
+        creativeAndUXPage.validateEmail("indira.gandhi@delhi.com");
+        creativeAndUXPage.validateCountry("India");
+        creativeAndUXPage.enterFirstName("Indira");
+        creativeAndUXPage.validateFirstName("Indira");
+        creativeAndUXPage.clickOnCotactUsButton();
+        creativeAndUXPage.assertLastNameWarningMessageIsShown();
```

Comment:
Test case states that for all the fields warning message should be shown, but there is only assertion for one. Why?

Reply:
I think that we already discussed about work logic of contact form, but if we are not we can arrange short meeting where I can explain.